src/installer/tests/HostActivation.Tests/MultiArchInstallLocation.cs

```diff
@@ -71,6 +71,56 @@ public void EnvironmentVariable_ArchSpecificDotnetRootIsUsedOverDotnetRoot()
                 .And.NotHaveStdErrContaining("Using environment variable DOTNET_ROOT=");
         }
 
+        [Fact]
+        public void EnvironmentVariable_DotnetRootPathDoesNotExist()
+        {
+            var fixture = sharedTestState.PortableAppFixture
+                .Copy();
+
+            var appExe = fixture.TestProject.AppExe;
+            using (TestOnlyProductBehavior.Enable(appExe))
+            {
+                Command.Create(appExe)
+                    .EnableTracingAndCaptureOutputs()
+                    .DotNetRoot("non_existent_path")
+                    .MultilevelLookup(false)
+                    .EnvironmentVariable(
+                        Constants.TestOnlyEnvironmentVariables.GloballyRegisteredPath,
+                        sharedTestState.InstallLocation)
+                    .Execute()
+                    .Should().Pass()
+                    .And.HaveStdErrContaining("Did not find [DOTNET_ROOT] directory [non_existent_path]")
+                    // If DOTNET_ROOT points to a folder that does not exist, we fall back to the global install path.
+                    .And.HaveUsedGlobalInstallLocation(sharedTestState.InstallLocation)
+                    .And.HaveStdOutContaining("Hello World");
+            }
+        }
+
+        [Fact]
+        public void EnvironmentVariable_DotnetRootPathExistsButHasNoHost()
+        {
+            var fixture = sharedTestState.PortableAppFixture
+                .Copy();
+
+            var appExe = fixture.TestProject.AppExe;
+            var projDir = fixture.TestProject.ProjectDirectory;
+            using (TestOnlyProductBehavior.Enable(appExe))
+            {
+                Command.Create(appExe)
+                .EnableTracingAndCaptureOutputs()
+                .DotNetRoot(projDir)
+                .MultilevelLookup(false)
+                .EnvironmentVariable(
+                    Constants.TestOnlyEnvironmentVariables.GloballyRegisteredPath,
+                    sharedTestState.InstallLocation)
+                .Execute()
+                .Should().Fail()
+                .And.HaveStdErrContaining($"Using environment variable DOTNET_ROOT=[{projDir}]")
+                // If DOTNET_ROOT points to a folder that exists we assume that there's a dotnet installation in it
+                .And.HaveStdErrContaining("A fatal error occurred. The required library hostfxr.dll could not be found.");
```

Comment:
> hostfxr.dll

There is `RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform` to get the platform-specific name.